hudhook/src/hooks/common.rs

```diff
@@ -0,0 +1,119 @@
+use imgui::*;
+use parking_lot::MutexGuard;
+use windows::Win32::Foundation::{HWND, LPARAM, LRESULT, WPARAM};
+use windows::Win32::UI::Input::KeyboardAndMouse::*;
+use windows::Win32::UI::WindowsAndMessaging::{WHEEL_DELTA, WM_XBUTTONDBLCLK, XBUTTON1, *};
+
+pub(crate) type WndProcType =
+    unsafe extern "system" fn(hwnd: HWND, umsg: u32, wparam: WPARAM, lparam: LPARAM) -> LRESULT;
+
+pub(crate) trait ImguiRendererCommon {
+    fn io_mut(&mut self) -> &mut imgui::Io;
+    fn set_focus(&mut self, focus: bool);
+    fn is_focus(&self) -> bool;
+    fn get_wnd_proc(&self) -> WndProcType;
+
+    fn init_io(&mut self) {
```

Comment:
```suggestion
    fn setup_io(&mut self) {
```
I'd rename this to `setup` for clarity, as it is actually the content which is initialized rather than the `Io` itself.